pkg/ddl/notifier/subscribe.go

```diff
@@ -255,7 +278,27 @@ func (n *DDLNotifier) processEventForHandler(
 	return nil
 }
 
-// Close releases the resources.
-func (n *DDLNotifier) Close() {
-	n.ownedSess.Close()
+// Stop stops the background loop.
+// Exposed for testing.
+// Do not call this function directly. Use owner.Listener interface instead.
+func (n *DDLNotifier) Stop() {
+	// If the notifier is not started, the cancel function is nil.
+	if n.cancel != nil {
+		n.cancel()
+		n.wg.Wait()
+	}
+}
+
+// OnBecomeOwner implements the owner.Listener interface.
+// We need to make sure only one DDLNotifier is running at any time.
+func (n *DDLNotifier) OnBecomeOwner() {
+	n.wg.RunWithRecover(n.Start, func(r any) {
+		logutil.BgLogger().Error("panic in ddl notifier", zap.Any("recover", r), zap.Stack("stack"))
```

Comment:
maybe use `intest` to let it really panic in our testing environment

catching the panic is friendly to other components in the same process to let them keep running, but it also mutes the error and make the bug hard to reveal.

Reply:
Updated.